x-pack/plugins/apm/server/routes/mobile/get_mobile_filters.ts

```diff
@@ -50,81 +50,134 @@ export async function getMobileFilters({
   start: number;
   end: number;
 }): Promise<MobileFiltersResponse> {
-  const response = await apmEventClient.search('get_mobile_filters', {
-    apm: {
-      events: [ProcessorEvent.transaction],
-    },
-    body: {
-      track_total_hits: false,
-      size: 0,
-      query: {
-        bool: {
-          filter: [
-            ...termQuery(SERVICE_NAME, serviceName),
-            ...termQuery(TRANSACTION_TYPE, transactionType),
-            ...rangeQuery(start, end),
-            ...environmentQuery(environment),
-            ...kqlQuery(kuery),
-          ],
+  async function getMobileTransactionEventsFilters(): Promise<
```

Comment:
Why do we need these 2 functions again ?
We already have this logic implemented in 
- `x-pack/plugins/apm/server/routes/mobile/get_mobile_most_used_charts/get_device_os_app_charts.ts`
- `x-pack/plugins/apm/server/routes/mobile/get_mobile_most_used_charts/get_nct_chart.ts`

The only difference is the `size` parameter which can be made available via parameter

Reply:
I'll have a look, but I rather have queries that search for the specific data we need instead of filtering out things and change parameters, either way we don't avoid the call to the different api